views/src/js/get_posts.js

```diff
@@ -0,0 +1,28 @@
+async function fetchPosts(filters = "") {
+    let url = 'http://localhost:8087/posts/'
+    if (filters !== "") {
+        url += "?" + filters;
+    }
+    try {
+        const response_data  = await fetch(url, {
+            method: 'GET',
+            mode: 'cors',
+            async: true,
+        });
+
+        if (response_data.status === 200) {
+            return response_data.json();
+        }
+    }
+    catch (e) {
+        console.error("No connection with server!")
```

Comment:
catch is good. by I meant create custom Error instance, and throw it for example, or  don't handle here but handle in component, which uses it